snap_plugin/v1/plugin.py

```diff
@@ -360,7 +376,18 @@ def start_plugin(self):
             sys.stdout.flush()
 
     def _generate_preamble_and_serve(self):
-        self._port = self.server.add_insecure_port('127.0.0.1:{!s}'.format(0))
+        if self.tls_enabled:
+            try:
+                self._TLS_setup()
+                credentials = self._generate_TLS_credentials()
+                self._port = self.server.add_secure_port('127.0.0.1:{}'.format(self._port), credentials)
+                LOG.info("Opened secure port on {}.".format(self._port))
+            except Exception as e:
+                LOG.error(str(e))
```

Comment:
This message is an information for plugin user, not necessary a developer. In my opinion, user shouldn't be presented with a traceback. Instead, we should log readable message (e.g. `Error reading root cert. Check file path and permissions.`). This will require distinction between different Exceptions, as mentioned in comment above.